addon/webextension/selector/ui.js

```diff
@@ -0,0 +1,627 @@
+/* globals window, document, console, browser */
+/* globals util, catcher, inlineSelectionCss */
+
+window.ui = (function () { // eslint-disable-line no-unused-vars
+  let exports = {};
+  const SAVE_BUTTON_HEIGHT = 50;
+
+  const { watchFunction } = catcher;
+
+  // The <body> tag itself can have margins and offsets, which need to be used when
+  // setting the position of the boxEl.
+  function getBodyRect() {
+    if (getBodyRect.cached) {
+      return getBodyRect.cached;
+    }
+    let rect = document.body.getBoundingClientRect();
+    let cached = {
+      top: rect.top + window.scrollY,
+      bottom: rect.bottom + window.scrollY,
+      left: rect.left + window.scrollX,
+      right: rect.right + window.scrollX
+    };
+    // FIXME: I can't decide when this is necessary
+    // *not* necessary on http://patriciogonzalezvivo.com/2015/thebookofshaders/
+    // (actually causes mis-selection there)
+    // *is* necessary on http://atirip.com/2015/03/17/sorry-sad-state-of-matrix-transforms-in-browsers/
+    cached = {top: 0, bottom: 0, left: 0, right: 0};
+    getBodyRect.cached = cached;
+    return cached;
+  }
+
+  exports.isHeader = function (el) {
+    while (el) {
+      if (el.classList &&
+          (el.classList.contains("myshots-button") ||
+           el.classList.contains("visible") ||
+           el.classList.contains("full-page"))) {
+        return true;
+      }
+      el = el.parentNode;
+    }
+    return false;
+  }
+
+  let substitutedCss = inlineSelectionCss.replace(/MOZ_EXTENSION([^\"]+)/g, (match, filename) => {
+    return browser.extension.getURL(filename);
+  });
+
+  function makeEl(tagName, className) {
+    if (! iframe.document()) {
+      throw new Error("Attempted makeEl before iframe was initialized");
+    }
+    let el = iframe.document().createElement(tagName);
+    if (className) {
+      el.className = className;
+    }
+    return el;
+  }
+
+  function onResize() {
+    if (this.sizeTracking.windowDelayer) {
+      clearTimeout(this.sizeTracking.windowDelayer);
+    }
+    this.sizeTracking.windowDelayer = setTimeout(watchFunction(() => {
+      this.updateElementSize(true);
+    }), 50);
+  }
+
+  let iframeSelection = exports.iframeSelection = {
+    element: null,
+    addClassName: "",
+    sizeTracking: {
+      timer: null,
+      windowDelayer: null,
+      lastHeight: null,
+      lastWidth: null
+    },
+    document: null,
+    display: function (installHandlerOnDocument) {
+      return new Promise((resolve, reject) => {
+        if (! this.element) {
+          this.element = document.createElement("iframe");
```

Comment:
If the iframe has a moz-extension: URL, your content script will be able to access its contents, since it has an expanded principal with a moz-extension: URL, but the content page won't.